 ## Profitable App Profiles for the App Store and Google Play Markets
 

```
***THE LITTLE PRETENDING STORY***
For this project, we'll pretend we're working as data analysts for a company that builds Android and iOS mobile apps. We make our apps available on Google Play and the App Store.
```

### Analyzing Mobile App Data

We only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means our revenue for any given app is mostly influenced by the number of users -- the more users that see and engage with the ads, the better.

Our goal for this project is to analyze data to help our developers understand **what type of apps are likely to attract more users.**

### Opening and Exploring the Data

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play, according to [Statica](https://www.statista.com/statistics/276623/number-of-apps-available-in-leading-app-stores/).

Collecting data for over 4 million apps requires a significant amount of time and money, so we'll try to analyze a sample of the data instead.

We have two data sets:
 - [A data set](https://www.kaggle.com/lava18/google-play-store-apps/home) containing approximately 10,000 Android apps (August 2018).
 - [A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home) containing approximately 7,000 iOS apps (July 2017).

In [51]:
from csv import reader
# The App Store data set
opened = open('AppleStore.csv', encoding='utf-8')
read = reader(opened)
ios = list(read)
ios_header = ios[0]
ios = ios[1:]

# The Google Play data set
opened = open('googleplaystore.csv', encoding='utf-8')
read = reader(opened)
    
android = list(read)
android_header = android[0]
android = android[1:]

In [52]:
# A function to explore the data
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n') #ads a new empty line after each row
        
    if rows_and_columns:
        print('Number of rows: ', len(dataset))
        print('Number of columns: ', len(dataset[0]))
        
# Android
print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows:  10841
Number of columns:  13


In [53]:
# iOS
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows:  7197
Number of columns:  16


### Deleting Wrong Data

We need to perform some data cleaning steps to have accurate data to analyze.
 - detect inaccurate data, adn correct or remove it.
 - detect duplicate data, and remove the duplicates.
 - remove apps that aren's free (because we are interested in analyzing free apps).
 
The Google Play data set has a dedicted [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) that describes an error fro the row 10472. Let's print the row and check whether it is indeed incorect.

In [54]:
print(android_header) # the header row
print('\n')
print(android[10472]) # the error row
print('\n')
print(android[0]) # a correct row  

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


The Category column is missing from the row 10472. We can delete this row.

In [55]:
print(len(android))
del android[10472] #!!! do not run this more than once
print(len(android))

10841
10840


### Removing Duplicate Entries
#### Part One

If we explore the Google Play data set long enough, we'll notice some apps have duplicate entries. For instance, Instagram has four entries:

In [56]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In total, there are 1,181 cases where an app occurs more than once:

In [57]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print('Number of duplicates apps: ', len(duplicate_apps))
print('\n')
print('Example of duplicate apps: ', duplicate_apps[:15])

Number of duplicates apps:  1181


Example of duplicate apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


We will build a criterion for removing the duplicates. The higher the number of reviews, the more recent the data should be. We will only keep the row with the highest number of reviews and remove the other entries for any given app.

To remove the duplicates, we will:
- create a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest number of review of that app.
- use the information stored in the dictionary and create a new data set, which will have only one entry per app (and for each app, we'll  only select the entry with the highest number of reviews).

#### Part Two

In [58]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews   

In [59]:
print('Expected entries number: ', len(android) - 1181)
print('Actual entries number: ', len(reviews_max))

Expected entries number:  9659
Actual entries number:  9659


<img src="img/happy.jpg" alt="happy" width="200">

Now we want to remove the duplicates using the *reviews_max* dicitionary.
- We start by initializing two empty lists, *android_clean* and *already_added*.
- We loop through the *android* data set, aand for every iteration:
    - we isolate the name of the app and the number of reviews.
    - we add the current app to the *android_clean* list, and the app name to the *already_cleaned* list if:
        - the number of reviews of the current app matches the number of reviews of that app as described in the dictionary.
        - the name of the app is not already in the *already_added* list.

In [60]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(app)
        already_added.append(name)

In [61]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows:  9659
Number of columns:  13


As we expected, we have 9659 rows!

### Removing Non-English Apps
#### Part One

We use English for the apps we develop at our company, and we'd like to analyze only the apps that are directed towards and English-speaking audience.

One way to go about this is to remove each app with a name containing a symbol that is not commonly used in English text.

Behind the scene, each character we use in a string has a corresponding number associated with it. We can get the corresponding number of each character using `ord()` built-in function.

According to [ASCII](https://en.wikipedia.org/wiki/ASCII), the numbers corresponding to the characters we commonly use in an English text are all in the range of 0 to 127.

In [62]:
def is_english(string):
    for character in string:
        if ord(character) > 127:
            return False
    return True
    
print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
False
False
False


In [63]:
print(ord('™'))

8482


We wrote a function that detects non-English app names, but we saw that the function couldn't correctly identify certain English app names because emojis and characters like `™` fall outside the ASCII range.

To minimize the impact of data loss, we'll only remove an an app if its name has more than three characters with corresponding numbers falling outside the ASCII range.

In [64]:
def is_english(string):
    count = 0
    for character in string:     
        if ord(character) > 127:
            count += 1
    if count > 3:
        return False
    else:
        return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
False
True
True


In [66]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 2, True)
print('\n')
explore_data(ios_english, 0, 2, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows:  9614
Number of columns:  13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows:  6183
Number of columns:  16


As a result of deleting some non-English apps we have 9614 rows of Android apps and 7197 of iOS apps.

By now, in the data cleaning process, we:
- Removed inaccurate data
- Removed duplicate app entries
- Removed non-English apps

As we mentioned at the begining, we only build free to download apps. The aim of the next step is to isolate the free apps and analyze the resulting data set.

### Isolating the Free Apps

In [67]:
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
        
explore_data(android_final, 0, 3, True)
print('\n')
explore_data(ios_final, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows:  8864
Number of columns:  13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+'

We have 8864 Android free apps left for analyzing and respectively, 3222 iOS free apps.

### Most Common Apps by Genre
#### Part One

Our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using the apps.

Our validation strategy for an app idea is comprised of three steps:
1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six motnhs, we build an iOS version of the app and add it to Apple Store.

Because the end goal is to add the app on both Google Play and App Store, we need to find app profiles that are successful on both markets. Let's begin the analysis by getting a sens of what are the most common genres for each market.

### Most Common Apps by Genre
#### Part Two

We need to build a frequency table for the `prime_genre` column of the App Store data, and for the `Genres` and `Category` columns of the Google Play data.

In [68]:
def freq_table(dataset, index):  #creates the frequency table
    table = {}
    total = 0
    
    for row in dataset:
        data_point = row[index]
        total += 1
        if data_point in table:
            table[data_point] += 1
        else:
            table[data_point] = 1
            
    table_pers = {}
    for key, value in table.items():
        table_pers[key] = value / total * 100
        
            
    return table_pers
            
print(android_header)
freq_table(android_final, -4)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


{'Art & Design': 0.5979241877256317,
 'Art & Design;Creativity': 0.06768953068592057,
 'Auto & Vehicles': 0.9250902527075812,
 'Beauty': 0.5979241877256317,
 'Books & Reference': 2.1435018050541514,
 'Business': 4.591606498194946,
 'Comics': 0.6092057761732852,
 'Comics;Creativity': 0.01128158844765343,
 'Communication': 3.2378158844765346,
 'Dating': 1.861462093862816,
 'Education': 5.347472924187725,
 'Education;Creativity': 0.04512635379061372,
 'Education;Education': 0.33844765342960287,
 'Education;Pretend Play': 0.056407942238267145,
 'Education;Brain Games': 0.033844765342960284,
 'Entertainment': 6.069494584837545,
 'Entertainment;Brain Games': 0.078971119133574,
 'Entertainment;Creativity': 0.033844765342960284,
 'Entertainment;Music & Video': 0.16922382671480143,
 'Events': 0.7107400722021661,
 'Finance': 3.7003610108303246,
 'Food & Drink': 1.2409747292418771,
 'Health & Fitness': 3.0798736462093865,
 'House & Home': 0.8235559566787004,
 'Libraries & Demo': 0.936371841155234

In [69]:
def display_table(dataset, index):   # helper function to display the frequency table in descending order
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

### Most Common Apps by Genre
#### Part Three

Let's analyze the iOS `prime_genre` column, and the Android `Genres` and `Category` columns.

In [70]:
print(ios_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [71]:
display_table(ios_final, -5) # iOS *prime_genre* column

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that more than 58% of the free English apps in the App Store are games, followed by Entertainment, which is almost 7.5 times less than the first one, with 7.8%. 

The data above gives the impression of high interest for fun categories like games, entertainment and social networking, and less for education, health, sports or other practical categories. However, this doesn't imply that the fun apps have more users -- the demand might not be the same as the offer.

Let's continue by analyzing the Google Store data.

In [72]:
print(android_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [73]:
display_table(android_final, 1) # Google Play *Category* column

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

We get a different picture for Google Play. Practical categories seem to come with a good offer (Family, Tools, Business) and less percentage for fun categories like Games. After further analysis, we will find that the Family category mostly includes games for children. Even so, the practical categories seem to have a better representation on Google Play than the App Store.

The 'Genres' column confirms this picture.

In [74]:
display_table(android_final, -4) # Google Play *Genres* column

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The frequency tables we analyze on the previous screen showed us that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and fun apps.

By calculating the average number of install for each app genre we can get an idea of the genre with most users.

For the App Store:
- Isolate the apps of each genre.
- Sum up the user ratings for the apps of the genre.
- Divide the sum by the number of apps belonging to the genre (not by the total number of apps).

### Most Popular Apps by Genre on the App Store

In [75]:
ios_genres = freq_table(ios_final, -5) # *prime_genre* column to get the unique app genres

for genre in ios_genres:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_ratings = total / len_genre
    print(genre, ':', avg_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


On average, navigation apps have the highest number of user reviews.

In [76]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


As we can see the Navigation category is influenced by Waze and Google Maps with almost half a million user reviews together.

The same pattern applies to Social Networking apps, where the average number is heavily influenced by a few giants like Facebook and Skype.

The aim is to find popular genres, but navigation, social networking or music apps are heavily influenced by a few outliers, they seem more popular than they really are.

The Reference category looks interesting.

In [77]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


Religious books and dictionaries have a big influence in the Reference category for App Store. Maybe coming up with a new great book can be a good idea.

The Weather category is very popular but people don't spend a long time on weather apps, except for specific areas of business, for instance, agriculture of construction. We are targeting many users so we can launch a free app.

Sports? Maybe a live streaming app. Soccer would definitely work for Europe.

Let's study the Google Play apps.

### Most Popular Apps by Genre on Google Play

In [78]:
display_table(android_final, 5) # the *Install* column

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


The install numbers don't seem precise enough -- we can see that most values are open-ended(100+, 1,000+, etc).

We don't know whether an app with 100+ installs has 100, 102 or 200. However, we don't need very precise data for our purpose -- we only want to find out which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We are going to leave the numbers as they are, which means we'll consider that an app with 100+ installs has 100 installs.

In [79]:
print(android_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [82]:
categories = freq_table(android_final, 1)

In [84]:
for category in categories:
    total = 0 
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace("+", "")
            n_installs = n_installs.replace(",", "")
            n_installs = float(n_installs)
            total += n_installs
            len_category += 1
            
    avg_installs = total / len_category
    print(category, ":", avg_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

On average, communication apps have the most installs.

In [85]:
for app in android_final:
    if app[1] == 'COMMUNICATION':
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
TracFone My Account : 1,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
My magenta : 1,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Seznam.cz : 1,000,000+
Antillean Gold Telegram (original version) : 100,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Calls & Text by Mo+ : 5,000,000+
free 

Here we have a few outliers that heavily influence the average number of installs: WhatsApp, Messanger, Skype. If we remove the outliers, the average number of installs would be reduce many times.

We see the same picture with Video Players category, the market is dominated by Youtube, Google Play Movies & TV, etc. the same pattern with Social apps (Facebook, Instagram, etc), photography apps (Google Photos).

In [86]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

Books category may be showing some potential on Google Play too.

In [87]:
for app in android_final:
    if app[1] == 'SPORTS':
        print(app[0], ':', app[5])

Free Sports TV : 100,000+
LiveScore: Live Sport Updates : 10,000,000+
Onefootball - Soccer Scores : 10,000,000+
Cristiano Ronaldo Wallpaper : 500,000+
Futbol24 : 1,000,000+
kicker football news : 5,000,000+
Football Live Scores : 5,000,000+
Pro 2018 - Series A and B : 5,000,000+
BeSoccer - Soccer Live Score : 10,000,000+
Sport.pl LIVE : 1,000,000+
The Team - Live Sport: football, tennis, rugby .. : 5,000,000+
MARCA - Sports Leader Diary : 5,000,000+
Sahadan Live Scores : 5,000,000+
Foot Mercato: transfers, results, news, live : 1,000,000+
All Football - Latest News & Videos : 10,000,000+
Eurosport : 10,000,000+
beIN SPORTS TR : 10,000,000+
All Football GO- Live Score, Games : 100,000+
Top Mercato: football news : 1,000,000+
GollerCepte Live Score : 1,000,000+
Championat : 1,000,000+
Hovercraft Racer : 100,000+
World Cup 2018 : 500,000+
GollerCepte 1903 : 500,000+
League18 : 500,000+
Speed Boat Racing : 1,000,000+
La Liga - Spanish Soccer League Official : 10,000,000+
Lionel Messi Wallp

### Conclusion

Taking a very popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store market. The markets are already full of libraries, so we need to add some special futures besides the raw version of the book. This might include daily quotes from the book, an audio version of the ook, quizzes, etc.

Sports might be a good niche too. Football live streaming with some futures added like sports bets, etc.